tests/transformers/test_numerical.py

```diff
@@ -33,6 +33,194 @@ def test_fit(self):
+        """Test fit rounding parameter with ``'auto'``
+
+        If the ``rounding`` parameter is set to ``'auto'``
+        and the data is very large, fit should learn
+        ``rounding_digits`` to be the biggest number of 0s
+        to round to that keeps the data the same.
+
+        Input:
+        - Array of data with numbers as big as 10^20
+        """
+        # Setup
+        exponents = [np.random.randint(10, 20) for i in range(10)]
+        big_numbers = [10**exponents[i] for i in range(10)]
+        data = np.array(big_numbers)
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan='nan', rounding='auto')
+        transformer.fit(data)
+
+        # Asserts
+        expected = -min(exponents) if min(exponents) > 0 else None
+        assert transformer.rounding_digits == expected
+
+    def test_fit_rounding_auto_max_decimals(self):
+        """Test fit rounding parameter with ``'auto'``
+
+        If the ``rounding`` parameter is set to ``'auto'``,
+        fit should learn the ``rounding_digits`` to be the max
+        number of decimal places seen in the data. The biggest
+        amount of decimals that can be accurately compared with
+        floats is 15.
+
+        Input:
+        - Array with a value that has over 15 decimals.
+        """
+        # Setup
+        data = np.array([0.000000000000001])
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan='nan', rounding='auto')
+        transformer.fit(data)
+
+        # Asserts
+        assert transformer.rounding_digits is None
+
+    def test_reverse_transform_rounding_none(self):
+        """Test ``reverse_transform`` when ``rounding`` is ``None``
+
+        The data should not be rounded at all.
+        """
+        # Setup
+        data = np.random.random(10)
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan=None)
+        transformer.rounding_digits = None
+        result = transformer.reverse_transform(data)
+
+        # Assert
+        assert np.array_equal(result, data)
+
+    def test_reverse_transform_rounding_positive_int(self):
+        """Test ``reverse_transform`` when ``rounding`` is a positive int
+
+        The data should round to the maximum number of decimal places
+        set in the ``rounding_digits`` value.
+
+        Input:
+        - Array with decimals
+
+        Output:
+        - Same array rounded to the provided number of decimal places
+        """
+        # Setup
+        data = np.array([1.1111, 2.2222, 3.3333, 4.44444, 5.555555])
+
+        # Run
+        transformer = NumericalTransformer(dtype=np.float, nan=None)
+        transformer.rounding_digits = 2
+        result = transformer.reverse_transform(data)
+
+        # Assert
+        expected_data = np.array([1.11, 2.22, 3.33, 4.44, 5.56])
+        assert np.array_equal(result, expected_data)
+
+    def test_reverse_transform_rounding_negative_int(self):
```

Comment:
We should distinguish between integer and float values in two separated tests, since they should produce different outputs.
Also notice that reverse transform should always get floats as input!

Reply:
The name here might be misleading. By `int` I meant that the argument is an `int`, not the input data